src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/OapEsIndex.scala

```diff
@@ -0,0 +1,255 @@
+              .endObject()
+            .endObject()
+          .endObject()
+        .endObject()
+      .endObject()
+  }
+
+  /**
+   * Drop the whole table index. Usually, we don't need this.
+   */
+  def dropESTableIndex(tableIndexName: String): Unit = {
+
+    if (checkEsTableIndexExists(tableIndexName)) {
+
+      client.admin().indices()
+        .delete(new DeleteIndexRequest(tableIndexName))
+        .actionGet()
+    } else {
+
+      logWarning(s"ES Index: $tableIndexName is not exist")
+    }
+  }
+
+  /**
+   * This is the function used to create OAP index. It may be called several time to build
+   * one OAP index file. We need to store value, rowId, indexFileName for each data record.
+   *
+   * Since we use ES Index to store all indexes for one table, we need an identifier to locate
+   * one specific OAP index file. So, data file name and column name is needed.
+   *
+   * @param records From data file, it's an array of (value, rowId) pair
+   * @param oapIndexFileName Index File name is the composition of indexName and dataFileName
+   * @param tableName table name (corresponding to one ES table index)
+   * @return successfully inserted row count
+   */
+  override def batchInsert(
+    records: Seq[(Key, Int)], oapIndexFileName: String, tableName: String): Int = {
+
+    val bulkRequest = client.prepareBulk()
+
+    records.foreach { case (key, rowId) =>
+      val obj = XContentFactory.jsonBuilder()
+        .startObject()
+        .field("key", key.getUTF8String(0))
+        .field("index_file_name", oapIndexFileName)
+        .field("rowId", rowId)
+        .endObject()
+      val indexRequest = client.prepareIndex(tableName, docType)
+        .setSource(obj).setId(rowId.toString)
+      bulkRequest.add(indexRequest)
+    }
+    bulkRequest.execute().actionGet()
+    client.admin().indices().refresh(new RefreshRequest(tableName)).actionGet()
+    records.length
+  }
+
+  def queryAllRecordIds(tableIndexName: String): Seq[String] = {
+    val queryBuilder = QueryBuilders.matchAllQuery()
+    val response = client.prepareSearch(tableIndexName)
+      .setTypes(docType)
+      .addDocValueField("rowId")
+      .setQuery(queryBuilder).setSize(100).execute().actionGet()
+
+    response.getHits.getHits.map(_.getId)
+  }
+
+  def queryIdsByIndexFileName(oapIndexFileName: String, tableIndexName: String): Seq[String] = {
+
+    val queryBuilder = QueryBuilders.termQuery("index_file_name", oapIndexFileName)
+    val response = client.prepareSearch(tableIndexName)
+      .setTypes(docType)
+      .addDocValueField("rowId")
+      .setQuery(queryBuilder).setSize(100).execute().actionGet()
+
+    response.getHits.getHits.map(_.getId)
+  }
+
+  /**
+   * Pick one idle connection from connection pool
+   * TODO: Should be a pool
+   *
+   * @return the idle ES TransportClient
+   */
+  override protected def getEsClient: TransportClient = {
+    val settings = Settings.builder
+      .put("cluster.name", "elasticsearch")
```

Comment:
cluster.name should be configurable, not all cluster is called `elasticsearch`,

Reply:
let's do this after all configs figured out. for example: host, port.